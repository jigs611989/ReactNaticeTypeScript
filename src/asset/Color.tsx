const Colors = {
  primary: '#f7a934',
  black: '#0B121E',
  white: '#FFFFFF',
  placeholder: '#747577',
  blueGreyBackground: 'rgba(11, 18, 30, 0.02)',
  blueGreyBorder: 'rgba(11, 18, 30, 0.1)',
  blueGrey: 'rgb(11, 18, 30)',
  darkText: '#101313',
}

export default Colors;
