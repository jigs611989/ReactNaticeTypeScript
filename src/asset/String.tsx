const Strings = {
  welcome: 'Welcome to Investec',
  enterYourName: 'Enter Your Name',
  buttonVariationTitle: '4 variations of a button',
  Buttons: {
    continue: 'Continue',
    buttonOption: 'Button Option Screen',
    pressMe: 'Press me',
    slideMe: 'Slide me to continue',
  },
  Alert: {
    invalidName: 'Please enter valid name',
    buttonOption1: 'First Button Pressed',
    buttonOption2: 'Second Button Pressed',
    buttonOption3: 'Third Button Pressed',
    buttonOption4: 'Unlocked!!',
  },
};

export default Strings;
