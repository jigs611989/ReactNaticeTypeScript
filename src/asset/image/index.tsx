const Images = {
  reactLogo: require('./ReactLoGo.png'),
};

export default Images;
