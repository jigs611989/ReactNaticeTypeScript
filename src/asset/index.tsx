export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
