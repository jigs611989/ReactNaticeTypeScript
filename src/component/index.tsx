export { default as Button } from './Button';
export { default as Label } from './Label';
export { default as UserName } from './UserName';
export { default as TextField } from './TextField';
export { default as SliderButton } from './SliderButton';
