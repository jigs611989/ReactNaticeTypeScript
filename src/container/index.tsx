export { default as WelcomeScreen } from './WelcomeScreen';
export { default as HomeScreen } from './HomeScreen';
export { default as ButtonOptionScreen } from './ButtonOptionScreen';
