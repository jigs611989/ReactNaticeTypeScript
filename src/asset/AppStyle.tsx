import { StyleSheet } from 'react-native';
import Color from './Color';
const AppStyle = StyleSheet.create({
  rootContainer: {
    flex: 1,
    backgroundColor: Color.black,
  },
  buttonStyle: {
    backgroundColor: Color.primary,
  },
});

export default AppStyle;
