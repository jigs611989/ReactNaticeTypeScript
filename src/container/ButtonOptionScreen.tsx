import React from 'react';
import { Alert, SafeAreaView, View } from 'react-native';
import { String } from '../asset';
import { Button, Label, SliderButton, UserName } from '../component';
import style from './style/ButtonOptionScreen';

const ButtonOptionScreen: React.FC = () => {
  return (
    <SafeAreaView style={[style.rootContainer, style.container]}>
      <UserName />
      <View style={style.spacer} />
      <Label title={String.buttonVariationTitle} style={style.labelStyle} />
      <Button
        testID={'ButtonOption1'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption1)}
        textStyle={style.buttonOption1}
      />
      <Button
        testID={'ButtonOption2'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption2)}
        style={style.buttonOption2}
      />
      <Button
        testID={'ButtonOption3'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption3)}
        style={style.buttonStyle}
      />
      <SliderButton
        testID={'ButtonOption4'}
        onPress={() => Alert.alert('', String.Alert.buttonOption4)}
      />
    </SafeAreaView>
  );
};

export default ButtonOptionScreen;
