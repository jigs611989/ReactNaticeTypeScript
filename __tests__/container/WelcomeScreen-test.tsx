import React from 'react';
import { Provider } from 'react-redux';
import { Alert } from 'react-native';
import configureStore from 'redux-mock-store';
import { WelcomeScreen } from '../../src/container';
import { render, fireEvent } from '@testing-library/react-native';
import { String } from '../../src/asset';

const mockStore = configureStore();

it('renders correctly', () => {
  const initialState = { name: '' };
  const store = mockStore(initialState);
  const { toJSON, getByText } = render(
    <Provider store={store}>
      <WelcomeScreen />
    </Provider>,
  );

  const continueButton = getByText(String.Buttons.continue);
  fireEvent.press(continueButton);
  expect(Alert.alert).toHaveBeenCalledTimes(1);

  expect(toJSON()).toMatchSnapshot();
});
