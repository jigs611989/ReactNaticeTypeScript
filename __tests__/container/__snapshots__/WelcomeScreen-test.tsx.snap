// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#0B121E",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "flex": 1,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "lineHeight": 28,
          "margin": 16,
          "marginBottom": 2,
        },
        undefined,
      ]
    }
  >
    Welcome to Investec
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "flex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgba(11, 18, 30, 0.1)",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "margin": 20,
          "padding": 16,
          "width": "80%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Enter Your Name"
      placeholderTextColor="#747577"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#101313",
            "flex": 1,
            "fontSize": 14,
            "lineHeight": 16,
            "padding": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f7a934",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Continue
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "flex": 1,
      }
    }
  />
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 12,
            "elevation": 1,
            "justifyContent": "center",
            "maxHeight": 740,
            "maxWidth": 340,
            "paddingHorizontal": 20,
            "paddingVertical": 30,
            "shadowColor": "#0B121E",
            "shadowOffset": Object {
              "height": -5,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
                "lineHeight": 28,
                "margin": 16,
                "marginBottom": 2,
              },
              Object {
                "color": "#0B121E",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginVertical": 9,
                "textAlign": "center",
              },
            ]
          }
        >
          Device Type
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
                "lineHeight": 28,
                "margin": 16,
                "marginBottom": 2,
              },
              Object {
                "color": "#747577",
                "fontSize": 16,
                "marginBottom": 32,
                "textAlign": "center",
              },
            ]
          }
        >
          You are running on 
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f7a934",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 1,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                },
                undefined,
              ]
            }
          >
            Okay
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
